src/sql/workbench/common/editor/query/queryEditorInput.ts

```diff
@@ -264,8 +269,36 @@ export abstract class QueryEditorInput extends EditorInput implements IConnectab
 	}
 
 	// Called to get the tooltip of the tab
-	public override getTitle(): string {
-		return this.getName(true);
+	public override getTitle(verbosity?: Verbosity): string {
+		let profile = this.connectionManagementService.getConnectionProfile(this.uri);
+		let fullTitle = '';
+		if (profile) {
+			let additionalOptions = this.connectionManagementService.getNonDefaultOptions(profile);
+			if (this._description && this._description !== '') {
+				fullTitle = this._description + ' ';
+			}
+			fullTitle += `${profile.serverName}`;
+			if (profile.databaseName) {
+				fullTitle += `.${profile.databaseName}`;
+			}
+			fullTitle += ` (${profile.userName || profile.authenticationType})`;
+
+			fullTitle += additionalOptions;
+		}
+		else {
+			fullTitle = this.getName(true);
+		}
+		switch (verbosity) {
+			case Verbosity.SHORT:
+				return this.getName(true);
+			case Verbosity.LONG:
+				// Used by tabsTitleControl as the tooltip hover.
+				return fullTitle;
+			default:
```

Comment:
Nit: `default` should appear before the `MEDIUM` case. And should could also be moved here since it's making the same action as `MEDIUM` and `defuaul`

Reply:
Already done so,